webapp/src/components/sidebar_right/sidebar_right.tsx

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import Scrollbars from 'react-custom-scrollbars';
+import {Theme} from 'mattermost-redux/types/preferences';
+import {makeStyleFromTheme, changeOpacity} from 'mattermost-redux/utils/theme_utils';
+
+import {RHSStates} from '../../constants';
+import GitlabItems from './gitlab_items';
+import {Item} from '../../types/gitlab_items';
+
+interface PropTypes {
+    username: string;
+    org: string;
+    gitlabURL: string;
+    reviews: Item[];
+    unreads: Item[],
+    yourPrs: Item[],
+    yourAssignments: Item[],
+    rhsState: string,
+    theme: Theme,
+    actions:any,
+};
+
+export function renderView(props: PropTypes) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--view'
+        />);
+}
+
+export function renderThumbHorizontal(props: PropTypes) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--horizontal'
+        />);
+}
+
+export function renderThumbVertical(props: PropTypes) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--vertical'
+        />);
+}
+
+
+function mapGitlabItemListToPrList(gilist: Item[]) {
+    if (!gilist) {
+        return [];
+    }
+
+    return gilist.map((pr: Item) => {
+        return {sha: pr.sha, project_id: pr.project_id, iid: pr.iid};
+    });
+}
+
+function shouldUpdateDetails(prs: Item[], prevPrs: Item[], targetState: string, currentState: string, prevState: string) {
+    if (currentState !== targetState) {
+        return false
+    }
+
+    if (currentState !== prevState) {
+        return true;
+    }
+
+    if (prs.length !== prevPrs.length) {
+        return true;
+    }
+
+    for (let i = 0; i < prs.length; i++) {
+        if (prs[i].id !== prevPrs[i].id) {
+            return true;
+        }
+    }
+}
+
+export default class SidebarRight extends React.PureComponent<PropTypes> {
```

Comment:
any particular reason why defining this component as a class component and not as functional component?

Reply:
There is no particular reason for using class component. The existing code was using class component so I also used class component.

Reply:
ok, this is ok then. However we always want to use functional/hooks based components. Not as part of this PR but for future ones, always that we can, let's go with that approach. Thanks.